src/rail/estimation/algos/deepdisc.py

```diff
@@ -72,86 +68,66 @@ def train(config, all_metadata, train_head=True):
-        schedulerHook = return_schedulerhook(optimizer)
-        hookList = [lossHook, schedulerHook, saveHook]
+        cfg.SOLVER.MAX_ITER = e1  # for DefaultTrainer
 
         trainer = return_lazy_trainer(
-            model, training_loader, optimizer, cfg, cfg_loader, hookList
+            model, training_loader, optimizer, cfg, hookList
         )
 
         trainer.set_period(print_frequency)
 
-        trainer.train(0, head_iters)
+        trainer.train(0, e1)
 
         if comm.is_main_process():
-            np.save(output_dir + output_name + "_losses", trainer.lossList)
-            # np.save(output_dir + output_name + "_val_losses", trainer.vallossList)
+            np.save(output_dir + run_name + "_losses", trainer.lossList)
+            # np.save(output_dir + run_name + "_val_losses", trainer.vallossList)
 
-        return model
 
-    else:
-        cfg.train.init_checkpoint = os.path.join(output_dir, output_name + ".pth")
-        cfg_loader.SOLVER.BASE_LR = 0.0001
-        cfg_loader.SOLVER.STEPS = [e2, e3]  # do not decay learning rate for retraining
-        cfg_loader.SOLVER.MAX_ITER = efinal  # for DefaultTrainer
 
-        model = instantiate(cfg.model)
-        model.to(cfg.train.device)
-        model = create_ddp_model(model, **cfg.train.ddp)
+    else:
+        cfg.train.init_checkpoint = os.path.join(output_dir, run_name + ".pth")
+        cfg.SOLVER.BASE_LR = 0.0001
+        cfg.SOLVER.MAX_ITER = efinal  # for DefaultTrainer
 
-        cfg.optimizer.params.model = model
         cfg.optimizer.lr = 0.0001
-        optimizer = solver.build_optimizer(cfg_loader, model)
-        cfg_loader.SOLVER.MAX_ITER = efinal  # for DefaultTrainer
 
-        saveHook = return_savehook(output_name)
-        lossHook = return_evallosshook(val_per, model, eval_loader)
-        schedulerHook = return_schedulerhook(optimizer)
-        hookList = [lossHook, schedulerHook, saveHook]
 
         trainer = return_lazy_trainer(
-            model, training_loader, optimizer, cfg, cfg_loader, hookList
+            model, training_loader, optimizer, cfg, hookList
         )
 
         trainer.set_period(print_frequency)
-        trainer.train(0, full_iters)
+        trainer.train(0, efinal)
 
         if comm.is_main_process():
-            losses = np.load(output_dir + output_name + "_losses.npy")
+            losses = np.load(output_dir + run_name + "_losses.npy")
             losses = np.concatenate((losses, trainer.lossList))
-            np.save(output_dir + output_name + "_losses", losses)
+            np.save(output_dir + run_name + "_losses", losses)
 
-            # vallosses = np.load(output_dir + output_name + "_val_losses.npy")
+            # vallosses = np.load(output_dir + run_name + "_val_losses.npy")
             # vallosses = np.concatenate((vallosses, trainer.vallossList))
-            # np.save(output_dir + output_name + "_val_losses", vallosses)
+            # np.save(output_dir + run_name + "_val_losses", vallosses)
```

Comment:
Will these lines need to be uncommented and conditionally executed when we make lossHook conditional?